numpy/ma/mrecords.py

```diff
@@ -198,8 +198,8 @@ def __getattribute__(self, attr):
         fielddict = ndarray.__getattribute__(self, 'dtype').fields
         try:
             res = fielddict[attr][:2]
-        except (TypeError, KeyError):
-            raise AttributeError("record array has no attribute %s" % attr)
+        except (TypeError, KeyError) as e:
+            raise AttributeError("record array has no attribute %s" % attr) from e
```

Comment:
This matches the following, so is fine, thanks!

https://github.com/numpy/numpy/blob/432b3899dea324d82e3b7aa79355934442f97cec/numpy/core/records.py#L464-L465